internal/neofs/neofs.go

```diff
@@ -246,6 +247,35 @@ func (x *NeoFS) DeleteContainer(ctx context.Context, id cid.ID, token *session.C
 	return nil
 }
 
+func (x *NeoFS) signMultipartObject(header *object.Object, signer neofscrypto.Signer, payloadHash hash.Hash) error {
+	var (
+		sig neofscrypto.Signature
+	)
+
+	header.SetPayloadChecksum(checksum.NewFromHash(checksum.SHA256, payloadHash))
+
+	if x.IsHomomorphicHashingEnabled() {
+		var homoChecksum = tz.New()
+
+		homoChecksum.Write(header.Payload())
+		header.SetPayloadHomomorphicHash(checksum.NewFromHash(checksum.TillichZemor, homoChecksum))
+	}
+
+	id, err := header.CalculateID()
+	if err != nil {
+		return fmt.Errorf("calculate ID: %w", err)
+	}
+	header.SetID(id)
+
+	if err = sig.Calculate(signer, id.Marshal()); err != nil {
+		return fmt.Errorf("sign object ID: %w", err)
+	}
+
+	header.SetSignature(&sig)
```

Comment:
just `CalculateAndSetID` and not 4 method calls above?

Reply:
It seems shorter, replaced

Reply:
@smallhive, my bad, i meant `SetIDWithSignature` but copied a different method name. it allows even fewer code lines for the same functionality

Reply:
This is even better, updated. We should make such changes in the Slicer as well. I inspired it's code